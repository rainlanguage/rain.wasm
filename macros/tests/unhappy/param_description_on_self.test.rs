#[macro_use]
extern crate wasm_bindgen_utils_macros;

struct TestStruct;

#[wasm_export]
impl TestStruct {
    #[wasm_export]
    pub fn test_method(
        #[wasm_export(param_description = "the instance")]
        &self,
        arg: String,
    ) -> Result<String, Error> {
        Ok(arg)
    }
}

fn main() {}