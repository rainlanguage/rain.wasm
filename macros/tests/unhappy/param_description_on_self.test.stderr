error: `param_description` cannot be used on receiver parameters (self, &self, &mut self)
  --> tests/unhappy/param_description_on_self.test.rs:10:9
   |
10 |         #[wasm_export(param_description = "the instance")]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
